llvm/lib/Object/DXContainer.cpp

```diff
@@ -228,6 +242,53 @@ void DXContainer::PartIterator::updateIteratorImpl(const uint32_t Offset) {
   IteratorState.Offset = Offset;
 }
 
+Error DirectX::RootSignature::parse(StringRef Data) {
+  const char *Current = Data.begin();
+
+  // Root Signature headers expects 6 integers to be present.
+  if (Data.size() < 6 * sizeof(uint32_t)) {
+    return parseFailed("Invalid data. Too small.");
```

Comment:
The error here should probably be more descriptive. Imagine having a tool print this error? Without context it doesn't really tell you what went wrong.

Maybe something more like "Invalid root signature, insufficient space for header."